app/views/dialogue.py

```diff
@@ -466,3 +459,118 @@ def _setup_messagebox(title: str | None) -> QMessageBox:
         dialogue.setWindowTitle(DEFAULT_TITLE)
 
     return dialogue
+
+
+def show_settings_error() -> None:
+    """
+    Displays a fatal error box indicating the app was
+    unable to start due to corrupt settings. Called if unable
+    to parse the settings file.
+
+    :param settings: settings model to allow resetting settings
+    """
+    logger.info("Showing settings failure box")
+    diag = SettingsFailureDialog()
+    diag.exec_()
+
+
+class SettingsFailureDialog(QDialog):
+    """Custom dialog to display fatal errors regarding settings parsing.
+
+    Has buttons to open the settings file, open the settings folder, or reset settings.
+    Exiting the dialog will terminate the application.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+
+        # Set up the message box
+        self.setWindowTitle("Unable to parse settings file!")
+        self.setModal(True)
+        self.setObjectName("dialogue")
+
+        # Dynamic sizing
+        self.setSizePolicy(
+            QSizePolicy(QSizePolicy.Policy.Expanding, QSizePolicy.Policy.Expanding)
+        )
+
+        # Add data
+        self.text = "Your RimSort settings file is corrupt.\nPlease choose one of the following options to proceed."
+
+        # Buttons
+        self.open_settings_file_btn = QPushButton("Open Settings File")
+        self.open_settings_folder_btn = QPushButton("Open Settings Folder")
+        self.reset_settings_btn = QPushButton("Reset Settings")
+        self.close_application_btn = QPushButton("Exit RimSort")
+
+        btn_layout = QHBoxLayout()
+        btn_layout.addWidget(self.open_settings_file_btn)
+        btn_layout.addWidget(self.open_settings_folder_btn)
+        btn_layout.addWidget(self.reset_settings_btn)
+
+        # Set up the layout
+        layout = QVBoxLayout()
+        main_layout = QHBoxLayout()
+        main_layout.setAlignment(Qt.AlignmentFlag.AlignLeft)
+
+        # Left-side
+        l_layout = _setup_error_icon(self)
+        main_layout.addLayout(l_layout)
+
+        # Center spacer
+        main_layout.addItem(QSpacerItem(10, 0))
+
+        # Right-side
+        r_layout = QVBoxLayout()
+
+        txt = QLabel(self.text)
+        txt.setWordWrap(True)
+        r_layout.addWidget(QLabel(self.text))
+
+        r_layout.addItem(QSpacerItem(20, 20))
+
+        r_layout.addLayout(btn_layout)
+
+        r_layout.addWidget(self.close_application_btn)
+        main_layout.addLayout(r_layout)
+
+        layout.addLayout(main_layout)
+
+        self.setLayout(layout)
+        self.setFixedWidth(self.sizeHint().width())
+
+        def _open_settings_file() -> None:
+            generic.platform_specific_open(
+                AppInfo().app_storage_folder / "settings.json"
```

Comment:
Wonder if its worth moving this into AppInfo() as a constant. May make things easier, and also if we want to in the future, make it easier to change the path.

Reply:
Done!